config/src/lib.rs

```diff
@@ -1,75 +1,113 @@
-    pub host: String,
-    pub p2p: Option<P2pConfig>,
-    pub port: u16,
+/// Connections-specific configuration
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub struct ConnectionsConfig {
+    #[serde(default = "ConnectionsConfig::default_server_addr")]
+    pub server_addr: SocketAddr,
+    #[serde(default = "ConnectionsConfig::default_inbound_limit")]
+    pub inbound_limit: u16,
+    #[serde(default = "ConnectionsConfig::default_outbound_limit")]
+    pub outbound_limit: u16,
+    #[serde(default = "ConnectionsConfig::default_known_peers")]
+    pub known_peers: Vec<SocketAddr>,
 }
 
-#[derive(Clone, Debug, Deserialize)]
-pub struct P2pConfig {
-    pub host: String,
-    pub port: u16,
+/// Storage-specific configuration
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub struct StorageConfig {
+    #[serde(default = "StorageConfig::default_db_path")]
+    pub db_path: String,
 }
 
-#[derive(Debug, Deserialize)]
-pub struct LoggingConfig {
-    pub file_log_level: Option<String>,
-    pub log_file_append: Option<bool>,
-    pub log_file_path: Option<String>,
-    pub log_to_file: Option<bool>,
-    pub log_to_stdout: Option<bool>,
-    pub stdout_log_level: Option<String>,
+impl Config {
+    fn default_connections() -> ConnectionsConfig {
+        ConnectionsConfig::default()
+    }
+
+    fn default_storage() -> StorageConfig {
+        StorageConfig::default()
+    }
+}
+
+impl Default for Config {
+    fn default() -> Config {
+        Config {
+            connections: Self::default_connections(),
+            storage: Self::default_storage(),
+        }
+    }
 }
 
-pub fn read_config() -> Option<Config> {
-    let name: String = String::from(CONFIG_FILE);
-    let mut input = String::new();
+impl ConnectionsConfig {
+    fn default_server_addr() -> SocketAddr {
+        "127.0.0.1:1234".parse().unwrap()
+    }
+
+    fn default_inbound_limit() -> u16 {
+        256
+    }
 
-    File::open(&name)
-        .and_then(|mut f| f.read_to_string(&mut input))
-        .unwrap();
+    fn default_outbound_limit() -> u16 {
+        8
+    }
+
+    fn default_known_peers() -> Vec<SocketAddr> {
+        Vec::default()
+    }
+}
 
-    match input.parse() {
-        Ok(toml) => {
-            let toml: Toml::Value = toml;
-            let toml_str = toml.to_string();
-            let decoded: Config = Toml::from_str(&toml_str).unwrap();
-            Some(decoded)
+impl Default for ConnectionsConfig {
+    fn default() -> ConnectionsConfig {
+        ConnectionsConfig {
+            server_addr: Self::default_server_addr(),
+            inbound_limit: Self::default_inbound_limit(),
+            outbound_limit: Self::default_outbound_limit(),
+            known_peers: Self::default_known_peers(),
         }
-        Err(error) => {
-            panic!("failed to parse TOML: {}", error);
+    }
+}
+
+impl StorageConfig {
+    fn default_db_path() -> String {
+        "witnet_db".to_string()
```

Comment:
This should be agreed with the team. In this other [PR](https://github.com/witnet/witnet-rust/pull/63), the default db path (added to the .gitignore) is `.wit`. One of them should prevail :)

Reply:
I have seen several `String`s while dealing with path, are there any reasons rejecting `std::path`?

Reply:
@csmoe I considered it but found out it is an unsized type and that I had to use it behind a `Box` and so I thought that maybe it was a bit too much. I'm going to give it a try again now that everything is more stable and see how it looks like :wink:

Reply:
@csmoe I tried again and now I remember, `Path::new` returns a reference type, that forces me to handle the lifetime and that makes my life more difficult so I decided to just go straight to the `String` type. Nevertheless if you want to give it a try if you think that with Path is going to be better.